src/acm_pca_cert_generator/certgen.py

```diff
@@ -0,0 +1,417 @@
+        certs (dict): An array of dicts containing aliases and certificate paths
+                      for SSL certs to add to the truststore, e.g.:
+                      [{"alias": "testcert", "cert": "/tmp/mycert.pem"}]
+
+    """
+    trusted_certs = []
+    for alias, cert in certs.items():
+        with open(cert) as f:
+            pem_cert = OpenSSL.crypto.load_certificate(
+                OpenSSL.crypto.FILETYPE_PEM, f.read()
+            )
+            asn_cert = OpenSSL.crypto.dump_certificate(
+                OpenSSL.crypto.FILETYPE_ASN1, pem_cert
+            )
+            trusted_certs.append(jks.TrustedCertEntry.new(alias, asn_cert))
+    keystore = jks.KeyStore.new("jks", trusted_certs)
+    keystore.save(truststore_path, truststore_password)
+
+
+def parse_trusted_cert_arg(trusted_cert_aliases, trusted_certs):
+    """Split the CLI arguments for trusted cert aliases and paths.
+
+    Args:
+        trusted_cert_aliases (str): comma-separated list of certificate aliases
+                                    to add to the truststore
+        trusted_certs (str): comma-separated list of certificates (paths) to add
+                             to the truststore
+
+    Returns:
+        list of dicts: A list of {"alias": "string", "cert": "string"} dicts
+                       containing a mapping of alias name to certifificate path
+                       for trusted certificates
+
+    """
+    aliases = trusted_cert_aliases.split(",")
+    cert_paths = trusted_certs.split(",")
+    certs = []
+    i = 0
+    for alias in aliases:
+        cert = {"alias": alias, "cert": cert_paths[i]}
+        certs.append(cert)
+        i += 1
+    return certs
+
+
+def _setupLogging(log_level):
+    level = logging.getLevelName(log_level.upper())
+    logger.setLevel(level)
+    boto3.set_stream_logger("boto3.resources", level)
+    logger.debug("Set level to {}".format(log_level))
+
+
+def _main(args):
+    args = parse_args(args)
+    _setupLogging(args.log_level)
+    key = generate_private_key(args.key_type, args.key_length)
+
+    subject_name_parts = ["C", "ST", "L", "O", "OU", "CN", "emailAddress"]
+    subject_details = {}
+    for name_part in subject_name_parts:
+        arg = "subject_{}".format(name_part.lower())
+        subject_details[name_part] = getattr(args, arg)
+
+    csr = generate_csr(key, args.key_digest_algorithm, subject_details)
+
+    client = boto3.client("acm-pca")
+    cert = sign_cert(
+        client, args.ca_arn, csr, args.signing_algorithm, args.validity_period
+    )
+    generate_keystore(
+        args.keystore_path,
+        args.keystore_password,
+        key,
+        cert,
+        args.key_alias,
+        args.private_key_password,
+    )
+
+    trusted_certs = parse_trusted_cert_arg(
+        args.trusted_cert_aliases, args.trusted_certs
+    )
+    generate_truststore(args.truststore_path, args.truststore_password, trusted_certs)
```

Comment:
`parse_trusted_cert_arg()` returns a list of dicts, but `generate_truststore()` is expecting a dict of dicts. One of the two needs changing to match